src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl

```diff
@@ -6,39 +6,100 @@ YARN_CACHE_DIR=/usr/local/share/yarn-cache
 if [ -d $YARN_CACHE_DIR ]
 then
 	echo
-    echo "Configuring Yarn cache folder ..."
+    echo "Configuring Yarn cache folder..."
     yarn config set cache-folder $YARN_CACHE_DIR
 fi
 
+zippedModulesFileName=node_modules.zip
+allModulesDirName=__oryx_all_node_modules
+prodModulesDirName=__oryx_prod_node_modules
+
+if [ "$ORYX_COPY_ONLY_PROD_MODULES_TO_OUTPUT" == "true" ]
+then
+	if [ -d $allModulesDirName ]
+	then
+		# Rename existing node_modules back to prod modules
+		mv node_modules $prodModulesDirName
```

Comment:
what if node_modules doesn't exist, like in the first build?

Reply:
Yeah, we could be more defensive here. I was depending on presence of `allModulesDirName` which would be present after one build.

Reply:
I made some changes in latest iteration which might look more complex but I think handle more scenarios correctly.